File: Sucelloz/app/src/main/java/com/sucelloztm/sucelloz/models/CategoriesWithSubCategories.java
package com.sucelloztm.sucelloz.models;

import androidx.room.Embedded;
import androidx.room.Relation;

import java.util.List;

/**
 * Link between category and subcategory entities
 */
public class CategoriesWithSubCategories {
    @Embedded public Categories categories;
    @Relation(
            parentColumn = "category_id",
            entityColumn = "categories_id"
    )
    public List<SubCategories> subCategories;
}

File: Fluz/app/src/main/java/com/example/fluz/data/relashionships/CategoriesWithSubCategories.kt
package com.example.fluz.data.relashionships

import androidx.room.Embedded
import androidx.room.Relation
import com.example.fluz.data.entities.Category
import com.example.fluz.data.entities.SubCategory

data class CategoriesWithSubCategories(
    @Embedded val category: Category,
    @Relation(
        parentColumn = "id",
        entityColumn = "categoryId",
    )
    val subCategories: List<SubCategory>
)
